### PR TITLE
drivers: sensor: lm75: add alert threshold support

### PR DESCRIPTION
**Pull request by blemouzy**
Add SENSOR_ATTR_ALERT and SENSOR_ATTR_HYSTERESIS attributes support. The code is heavily inspired by the one for lm77 sensor.

### CODE REVIEW

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Now that this sensor supports triggers, please add it to `tests/drivers/build_all/sensor/sensors_trigger_*.conf` to ensure it builds with triggers enabled.

**Comment by blemouzy** [comment]
> Now that this sensor supports triggers, please add it to `tests/drivers/build_all/sensor/sensors_trigger_*.conf` to ensure it builds with triggers enabled.

Done!

**Review by MaureenHelm — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
Please rebase to trigger a new CI run

**Review by MaureenHelm — APPROVED** [review]
> Please rebase to trigger a new CI run

Sorry, one more time please. There are a couple of checks that never returned.

**Comment by MaureenHelm** [comment]
@teburd @yperess please take a look

File: drivers/sensor/lm75/Kconfig [hunk 1]
```diff
@@ -10,4 +10,43 @@ config LM75
...
+	help
+	  Specify the type of triggering to be used by the driver.
+
+config LM75_TRIGGER_NONE
+	bool "No trigger"
+
+config LM75_TRIGGER_GLOBAL_THREAD
+	bool "Use global thread"
+	depends on GPIO
+	select LM75_TRIGGER
```
Comment (ubieda) [inline drivers/sensor/lm75/Kconfig:29]: Not sure on the exact syntax, but the idea is that we don't enable trigger support unless we have `int-gpios` defined.
```suggestion
config LM75_TRIGGER_GLOBAL_THREAD
	bool "Use global thread"
	depends on $(dt_compat_any_has_prop,$(DT_COMPAT_LM75),int-gpios))
	select LM75_TRIGGER
```

File: drivers/sensor/lm75/lm75.c [hunk 2]
```diff
@@ -127,6 +339,42 @@ int lm75_init(const struct device *dev)
...
+
+		gpio_init_callback(&data->int_gpio_cb, lm75_int_gpio_callback_handler,
+				   BIT(cfg->int_gpio.pin));
+
+		ret = gpio_add_callback(cfg->int_gpio.port, &data->int_gpio_cb);
+		if (ret < 0) {
+			LOG_ERR("failed to add INT GPIO callback (ret %d)", ret);
+			return ret;
+		}
+	}
```
Comment (ubieda) [inline drivers/sensor/lm75/lm75.c:375]: It does not seem like this driver will work with `LM75_TRIGGER_SUPPORT` enabled and without `int_gpio` pin defined, yet there's a path to gracefully initialize in that scenario.

```suggestion
		if (!device_is_ready(cfg->int_gpio.port)) {
			LOG_ERR("INT GPIO not ready");
			return -EINVAL;
		}

		ret = gpio_pin_configure_dt(&cfg->int_gpio, GPIO_INPUT);
		if (ret < 0) {
			LOG_ERR("failed to configure INT GPIO (ret %d)", ret);
			return ret;
		}

		gpio_init_callback(&data->int_gpio_cb, lm75_int_gpio_callback_handler,
				   BIT(cfg->int_gpio.pin));

		ret = gpio_add_callback(cfg->int_gpio.port, &data->int_gpio_cb);
		if (ret < 0) {
			LOG_ERR("failed to add INT GPIO callback (ret %d)", ret);
			return ret;
		}
```

**Review by ubieda — APPROVED** [review]
There's a loose end that needs addressed (int-gpios and Trigger Kconfig setting). I'm, however, approving this PR in pro of getting it in for the next release, as it's not critical.

If it does end up getting merged, I'll capture a low-priority bug to get this addressed.

File: drivers/sensor/lm75/lm75.c (hunk 2 above)
Comment (ubieda) [reply drivers/sensor/lm75/lm75.c:375]: #74342 74342

File: drivers/sensor/lm75/Kconfig (hunk 1 above)
Comment (ubieda) [reply drivers/sensor/lm75/Kconfig:29]: #74342

File: drivers/sensor/lm75/lm75.c (hunk 2 above)
Comment (ubieda) [reply drivers/sensor/lm75/lm75.c:375]: As pointed out here https://github.com/zephyrproject-rtos/zephyr/pull/74342#discussion_r1645904160, this suggestion wasn't accurate and won't go in the follow-up PR.

**Pull request merged**